From 5c3e203ce54c61a8aa6e76be99cd83169c8da5ca Mon Sep 17 00:00:00 2001
From: Adrian-Ken Rueegsegger <ken@codelabs.ch>
Date: Tue, 21 Jan 2020 13:53:59 +0100
Subject: [PATCH 1/3] mutime: Make Sinfo an imported constant

The subject info data structure is static and is not changed during
runtime. Also adjust refinement of abstract timer state and adjust
global contract of Hz function since it does not need to be volatile.

Signed-off-by: Adrian-Ken Rueegsegger <ken@codelabs.ch>
---
 common/hw-time-timer.ads        | 4 +---
 common/mutime/hw-time-timer.adb | 5 ++---
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/common/hw-time-timer.ads b/common/hw-time-timer.ads
index 90978df..2bec99b 100644
--- a/common/hw-time-timer.ads
+++ b/common/hw-time-timer.ads
@@ -15,8 +15,7 @@
 private package HW.Time.Timer
    with
       Abstract_State => ((Timer_State with
-                           Part_Of => HW.Time.State,
-                           External => Async_Writers),
+                           Part_Of => HW.Time.State),
                          (Abstract_Time with
                            Part_Of => HW.Time.State,
                            External => Async_Writers))
@@ -38,7 +37,6 @@ is
 
    function Hz return T
    with
-      Volatile_Function,
       Global => (Input => Timer_State);
 
 end HW.Time.Timer;
diff --git a/common/mutime/hw-time-timer.adb b/common/mutime/hw-time-timer.adb
index c683d6f..8e9b497 100644
--- a/common/mutime/hw-time-timer.adb
+++ b/common/mutime/hw-time-timer.adb
@@ -25,10 +25,9 @@ is
      := ((Musinfo.Subject_Info_Type_Size + (16#1000# - 1))
          / 16#1000#) * 16#1000#;
 
-   Sinfo : Musinfo.Subject_Info_Type
+   Sinfo : constant Musinfo.Subject_Info_Type
    with
-      Volatile,
-      Async_Writers,
+      Import,
       Address => System'To_Address (Sinfo_Base_Address);
 
    Sched_Info : Muschedinfo.Scheduling_Info_Type
-- 
2.20.1

